File: Week3/Homework/hw3.1/hw3-1.js
db.students.find().forEach(
  function(student) {
    tmp_score = {type: 'homework', score: 0};
    new_scores = [];
    for(y=0; y<student.scores.length; y++) {
      if (student.scores[y].type == 'homework' && student.scores[y].score > tmp_score.score) {
        tmp_score = student.scores[y];
      }
      else if (student.scores[y].type != 'homework') {
        new_scores.push(student.scores[y]);
      }
    }
    new_scores.push(tmp_score);
    db.students.update({_id:student._id}, {$set: {scores: new_scores}});
  }
)

File: Week3/Homework/hw3.1/hw3-1.rb
require 'rubygems'
require 'mongo'

include Mongo

@client = MongoClient.new('localhost', 27017)
@db     = @client['school']
@coll   = @db['students']

@coll.find().each { |doc|

  # Finds the lowest scores id
  d = nil
  doc['scores'].each_index { |i|
    d = i if doc['scores'][i]['type'] == 'homework' && (d.nil? || doc['scores'][i]['score'] < doc['scores'][d]['score'])
  }

  # Removes the lowest score
  doc['scores'].delete_at(d) if !d.nil?

  # Updates the scores array
  @coll.update({_id: doc['_id']}, {'$set' => {scores: doc['scores']}})

}
